### ISSUE TITLE
Change ShortCode arguments to ILookup<string, string>

### ISSUE THREAD

**Issue by patriksvensson** [statiqdev/Statiq.Framework#97]
First, thanks for a great project! I really enjoying working with it.

This is not really an issue but rather a minor suggestion.
I would suggest changing the API for ShortCode from 

```csharp
async Task<IEnumerable<ShortcodeResult>> IShortcode.ExecuteAsync(
            KeyValuePair<string, string>[] args,
            string content,
            IDocument document,
            IExecutionContext context)
```

to

```csharp
async Task<IEnumerable<ShortcodeResult>> IShortcode.ExecuteAsync(
            ILookup<string, string> args,
            string content,
            IDocument document,
            IExecutionContext context)
```

This would make it a little bit ergonomic working with arguments.

**Comment by daveaglick** [comment]
Good idea - I didn’t want to use a dictionary since shortcodes can have more than one value for a given argument. A lookup could work well though, hadn’t thought of that.

Did you see the `args.SingleValue()` and `args.ToDictionary()` extensions? Those help convert the args to a value or metadata dictionary - if I move to a lookup, I’ll change the extensions to operate on that.

**Comment by patriksvensson** [comment]
@daveaglick Did not see the extensions, but will do a `.ToDictionary().ToLookup()` until (if) the `ILookup` change gets implemented. 👍
